FeedSX/screens/createPost/index.tsx

```diff
@@ -0,0 +1,549 @@
+      if (response?.didCancel) {
+        console.log('didCancel', response.didCancel);
+      }
+      let selectedImages = response?.assets; // selectedImages can be anything images or videos or both
+
+      for (let i = 0; i < selectedImages?.length; i++) {
+        if (selectedImages[i].fileSize >= MAX_FILE_SIZE) {
+          console.log('Files above 100 MB is not allowed');
+          return;
+        }
+      }
+      if (!!selectedImages) {
+        convertData(selectedImages);
+        setShowOptions(false);
+      }
+    });
+  };
+
+  //select Documents From Gallery
+  const selectDoc = async () => {
+    try {
+      const response = await DocumentPicker.pick({
+        type: [DocumentPicker.types.pdf],
+        allowMultiSelection: true,
+      });
+      let selectedDocs: any = response; // selectedImages can be anything images or videos or both
+      let docsArrlength = selectedDocs?.length;
+      if (docsArrlength > 0) {
+        for (let i = 0; i < docsArrlength; i++) {
+          if (selectedDocs[i].size >= MAX_FILE_SIZE) {
+            console.log('Files above 100 MB is not allowed');
+            return;
+          }
+        }
+        if (!!selectedDocs) {
+          convertDocData(selectedDocs);
+          setShowOptions(false);
+        }
+      }
+    } catch (error) {
+      console.log('err', error);
+    }
+  };
+
+  // function handles the selection of images and videos
+  const handleGallery = async (type: string) => {
+    if (Platform.OS === 'ios') {
+      selectGalley(type);
+    } else {
+      let res = await requestStoragePermission();
+      if (res === true) {
+        selectGalley(type);
+      }
+    }
+  };
+
+  // function handles the slection of documents
+  const handleDoc = async () => {
+    if (Platform.OS === 'ios') {
+      selectDoc();
+    } else {
+      let res = await requestStoragePermission();
+      if (res === true) {
+        selectDoc();
+      }
+    }
+  };
+
+  // function removes the selected documents
+  const removeDocAttachment = (index: number) => {
+    let newDocAttachments = [...formattedDocAttachments];
+    if (formattedDocAttachments.length === 1) {
+      setFormattedDocAttachments([]);
+      setShowOptions(true);
+    } else {
+      newDocAttachments.splice(index, 1);
+      setFormattedDocAttachments(newDocAttachments);
+    }
+  };
+
+  // function removes multiple images/videos selected
+  const removeMediaAttachment = (index: number) => {
+    let newMediaAttachments = [...formattedMediaAttachments];
+    newMediaAttachments.splice(index, 1);
+    setFormattedMediaAttachments(newMediaAttachments);
+  };
+
+  // function removes single image and video selected
+  const removeSingleAttachment = () => {
+    setFormattedMediaAttachments([]);
+    setShowOptions(true);
+  };
+
+  // function finds the valid links
+  const detectLinks = (text: string) => {
+    const regex = /\b(?:https?:\/\/)?(?:[\w.]+\.\w+)(?:(?<=\\n)|\b)/;
```

Comment:
put all the regex in a separate file and import from there

Reply:
resolved